src/main/java/org/opensearch/neuralsearch/sparse/algorithm/BatchClusteringTask.java

```diff
@@ -97,7 +99,8 @@ public List<Pair<BytesRef, PostingClusters>> get() {
                 );
                 List<DocumentCluster> clusters = postingClustering.cluster(docFreqs);
                 postingClusters.add(Pair.of(term, new PostingClusters(clusters)));
-                InMemoryClusteredPosting.InMemoryClusteredPostingWriter.writePostingClusters(key, term, clusters);
```

Comment:
the logic in line97 doesn't write data back to cache, please check if there are other similar code which misses cache gated solution